crates/fluvio-compression/src/lib.rs

```diff
@@ -0,0 +1,109 @@
+use std::str::FromStr;
+use std::mem;
+
+mod error;
+#[cfg(feature = "gzip")]
+mod gzip;
+#[cfg(feature = "snappy")]
+mod snappy;
+
+#[cfg(feature = "lz4")]
+mod lz4;
+
+pub use error::CompressionError;
+use serde::{Serialize, Deserialize};
+
+/// The compression algorithm used to compress and decompress records in fluvio batch
+#[derive(Copy, Clone, Debug, PartialEq, Serialize, Deserialize)]
+#[serde(rename_all = "lowercase")]
+#[repr(i8)]
+pub enum Compression {
+    None = 0,
+
+    #[cfg(feature = "gzip")]
```

Comment:
Since compression type is configurable per topic, it should be dynamic not compiler configuration

Reply:
It is not clear, to me. Should we add an attribute to topic configuration to decide which compression algorithm the producer should use?

I thought  that is just responsibility of the producer to decide which compression algorithm to use

Reply:
I removed feature flags in this crate